lib/bloc/buy_bitcoin/moonpay/moonpay_bloc.dart

```diff
@@ -34,9 +34,16 @@ class MoonPayBloc extends Cubit<MoonPayState> {
     }
 
     try {
-      final url = await _breezLib.buyBitcoin(sdk.BuyBitcoinProvider.Moonpay);
-      _log.v("fetchMoonpayUrl url: $url");
-      emit(MoonPayState.urlReady(url));
+      if (swapInfo!.channelOpeningFees != null) {
+        sdk.BuyBitcoinRequest reqData = sdk.BuyBitcoinRequest(
+          provider: sdk.BuyBitcoinProvider.Moonpay,
+          openingFeeParams: swapInfo.channelOpeningFees,
+        );
+        final buyBitcoinResponse = await _breezLib.buyBitcoin(reqData: reqData);
+        final url = buyBitcoinResponse.url;
+        _log.v("fetchMoonpayUrl url: $url");
+        emit(MoonPayState.urlReady(url));
+      }
```

Comment:
As part of **TODO:2**,
`MoonPayState` will also contain the `openingFeeParams` that was used during the creation of this `url`.